### PR TITLE
Cancel all

### PR DESCRIPTION
**Pull request by zarazek**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ apply plugin: 'maven'
 apply plugin: 'signing'
 
 group 'net.quedex'
-version '0.7.0-SNAPSHOT'
+version '0.8.0-SNAPSHOT'
```
Comment (wgromniak) [inline build.gradle]: Version shouldn't be bumped. Please revert.

File: src/main/java/net/quedex/api/user/OrderListener.java [hunk 2]
```diff
@@ -12,6 +12,10 @@
 
     void onOrderCancelFailed(OrderCancelFailed orderCancelFailed);
 
+    void onAllOrdersCancelled();
+
+    void onCancelAllOrdersFailed(CancelAllOrdersFailedCause cause);
```
Comment (wgromniak) [inline src/main/java/net/quedex/api/user/OrderListener.java]: I think it should be wrapped in `CancelAllOrdersFailed` - we won't be able to extend it easily, e.g. with per-instrument or per-instrument-class cancellation.

File: src/main/java/net/quedex/api/user/UserMessageReceiver.java [hunk 3]
```diff
@@ -167,6 +173,18 @@ private void onOrderCancelFailed(OrderCancelFailed orderCancelFailed) {
         }
     }
 
+    private void onAllOrdersCanceled() {
+        if (orderListener != null) {
```
Comment (wgromniak) [inline src/main/java/net/quedex/api/user/UserMessageReceiver.java]: Please follow the pattern from other methods. There is a race condition between checking `null` and calling the listener.

File: build.gradle (hunk 1 above)
Comment (zarazek) [reply build.gradle]: OK

File: src/main/java/net/quedex/api/user/OrderListener.java (hunk 2 above)
Comment (zarazek) [reply src/main/java/net/quedex/api/user/OrderListener.java]: OK

File: src/main/java/net/quedex/api/user/UserMessageReceiver.java (hunk 3 above)
Comment (zarazek) [reply src/main/java/net/quedex/api/user/UserMessageReceiver.java]: OK

**Comment by zarazek** [comment]
All comments addressed

**Pull request merged**